tests/tests/checks/snapshots/analysis/cpp/bug1781178.cpp/check_glob@WithOverloads_Overloaded.snap

```diff
@@ -0,0 +1,129 @@
+---
+source: tests/test_check_insta.rs
+expression: "&json_results"
+snapshot_kind: text
+---
+[
+  {
+    "loc": "00076:14-24",
+    "source": 1,
+    "syntax": "decl,function",
+    "type": "void (int)",
+    "pretty": "function WithOverloads::Overloaded",
+    "sym": "_ZN13WithOverloads10OverloadedEi"
+  },
+  {
+    "loc": "00076:14-24",
+    "target": 1,
+    "kind": "decl",
+    "pretty": "WithOverloads::Overloaded",
+    "sym": "_ZN13WithOverloads10OverloadedEi",
+    "context": "WithOverloads",
+    "contextsym": "T_WithOverloads",
+    "peekRange": "76-76"
+  },
+  {
+    "loc": "00077:14-24",
+    "source": 1,
+    "syntax": "decl,function",
+    "type": "void (float)",
+    "pretty": "function WithOverloads::Overloaded",
+    "sym": "_ZN13WithOverloads10OverloadedEf"
+  },
+  {
+    "loc": "00077:14-24",
+    "target": 1,
+    "kind": "decl",
+    "pretty": "WithOverloads::Overloaded",
+    "sym": "_ZN13WithOverloads10OverloadedEf",
+    "context": "WithOverloads",
+    "contextsym": "T_WithOverloads",
+    "peekRange": "77-77"
+  },
+  {
+    "loc": "00078:14-24",
+    "source": 1,
+    "syntax": "decl,function",
+    "type": "void (_Bool)",
+    "pretty": "function WithOverloads::Overloaded",
+    "sym": "_ZN13WithOverloads10OverloadedEb"
+  },
+  {
+    "loc": "00078:14-24",
+    "target": 1,
+    "kind": "decl",
+    "pretty": "WithOverloads::Overloaded",
+    "sym": "_ZN13WithOverloads10OverloadedEb",
+    "context": "WithOverloads",
+    "contextsym": "T_WithOverloads",
+    "peekRange": "78-78"
+  },
+  {
+    "loc": "00082:4-14",
+    "source": 1,
+    "syntax": "use,function",
+    "type": "void (_Bool)",
+    "pretty": "function WithOverloads::Overloaded",
+    "sym": "_ZN13WithOverloads10OverloadedEb",
+    "confidence": [
+      "cppTemplateHeuristic"
+    ]
+  },
+  {
+    "loc": "00082:4-14",
+    "source": 1,
+    "syntax": "use,function",
+    "type": "void (float)",
+    "pretty": "function WithOverloads::Overloaded",
+    "sym": "_ZN13WithOverloads10OverloadedEf",
+    "confidence": [
+      "cppTemplateHeuristic"
+    ]
+  },
+  {
+    "loc": "00082:4-14",
+    "source": 1,
+    "syntax": "use,function",
+    "type": "void (int)",
+    "pretty": "function WithOverloads::Overloaded",
+    "sym": "_ZN13WithOverloads10OverloadedEi",
+    "confidence": [
+      "cppTemplateHeuristic"
+    ]
+  },
+  {
+    "loc": "00082:4-14",
+    "source": 1,
+    "syntax": "use,function",
+    "type": "void (int)",
+    "pretty": "function WithOverloads::Overloaded",
+    "sym": "_ZN13WithOverloads10OverloadedEi"
+  },
```

Comment:
ah, okay, so we see it both as a heuristic confidence and with an (implicit) concrete confidence

Reply:
Here yes, because this file doesn't go through merger.rs. (Only the files listed in $MULTI_CPP_FILES in `tests/tests/build` go through merger.rs.)
If it did, we would only keep the second entry (the one without cppTemplateHeuristic). format.rs also has some merge logic and only one entry is emitted in `data-symbols` and `data-confidences`.